tests/discovery.spec.coffee

```diff
@@ -53,6 +54,18 @@ describe 'Discoverable Services:', ->
 			if dummyService.tags?
 				expect(service.tags).to.deep.equal(dummyService.tags)
 
+	inspectPublishedAddresses = (service, hostIPs) ->
+		if hostIPs?
+			ips = hostIPs
+		else
+			interfaces = os.networkInterfaces()
+			ips = []
+			_.forEach _.flatten(_.values(interfaces)), (value) ->
```

Comment:
I think this would be nicer as
```coffee
ips = _(interfaces)
	.values()
	.flatten()
	.reject('internal')
	.map('address')
	.value()
```
(and lodash will use shortcut fusion to combine the `reject` and `map` into just one iteration for us, so we don't lose out there either)